app/src/androidTest/java/com/epfl/drawyourpath/database/MockDatabaseTest.kt

```diff
@@ -0,0 +1,520 @@
+            nonExistingStartingTime,
+            nonExistingStartingTime + 2e6.toLong(),
+        )
+        database.removeRunFromHistory(userIdTest, nonExistingRun)
+
+        val expectedHistory = listOf<Run>()
+
+        assertEquals(
+            expectedHistory.size,
+            database.users[userIdTest]?.runs?.size,
+        )
+    }
+
+    /**
+     * Test if adding a dailyGoal in the database is correctly made
+     */
+    @Test
+    fun addDailyGoalCorrect() {
+        val database = MockDatabase()
+        database.addDailyGoal(
+            userIdTest,
+            DailyGoal(
+                25.0,
+                30.0,
+                2,
+                20.0,
+                120.0,
+                1,
+                LocalDate.of(2010, 1, 1),
+            ),
+        ).get()
+        /*
+
+        // control the dailyGoal List
+        val obtainedDailyGoalList =
+            database.users[database.userIdTest]!!.getDailyGoalList()
+        assertEquals(obtainedDailyGoalList.size, 2)
+
+        // check the first daily goal
+        assertEquals(obtainedDailyGoalList.get(0).date, database.dailyGoalListTest.get(0).date)
+        assertEquals(
+            obtainedDailyGoalList.get(0).expectedDistance,
+            database.dailyGoalListTest.get(0).expectedDistance,
+            0.001,
+        )
+        assertEquals(
+            obtainedDailyGoalList.get(0).expectedTime,
+            database.dailyGoalListTest.get(0).expectedTime,
+            0.001,
+        )
+        assertEquals(
+            obtainedDailyGoalList.get(0).expectedPaths,
+            database.dailyGoalListTest.get(0).expectedPaths,
+        )
+        assertEquals(
+            obtainedDailyGoalList.get(0).distance,
+            database.dailyGoalListTest.get(0).distance,
+            0.001,
+        )
+        assertEquals(
+            obtainedDailyGoalList.get(0).time,
+            database.dailyGoalListTest.get(0).time,
+            0.001,
+        )
+        assertEquals(
+            obtainedDailyGoalList.get(0).paths,
+            database.dailyGoalListTest.get(0).paths,
+        )
+
+        // check the second daily goal
+        assertEquals(obtainedDailyGoalList.get(1).date, LocalDate.of(2010, 1, 1))
+        assertEquals(obtainedDailyGoalList.get(1).expectedDistance, 25.0, 0.001)
+        assertEquals(obtainedDailyGoalList.get(1).expectedTime, 30.0, 0.001)
+        assertEquals(obtainedDailyGoalList.get(1).expectedPaths, 2)
+        assertEquals(obtainedDailyGoalList.get(1).distance, 20.0, 0.001)
+        assertEquals(obtainedDailyGoalList.get(1).time, 120.0, 0.001)
+        assertEquals(obtainedDailyGoalList.get(1).paths, 1)
+        */
+    }
+
+    /**
+     * Test if the achievements are update correctly
+     */
+    @Test
+    fun updateUserAchievementsCorrect() {
+        val database = MockDatabase()
+        /*
+        database.updateUserAchievements(userIdTest, 10.0, 50.0).get()
```

Comment:
`updateUserAchievements` is not implemented yet, so I can't test it.

Reply:
I have discussed with @HoLuDyEr and we will need the achievements for the trophies and we will also used them to track the past performance of the user(to have some statistic). That is why I have implemented the achievements the last sprint. So it could be great if we can keep them since thy are totally implemented in the FireBase and MockDataBase and tested.